enterprise/frontend/src/embedding-sdk/hooks/private/use-sdk-usage-problem.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useMemo, useRef } from "react";
+import { useDispatch } from "react-redux";
+
+import type { SDKConfig } from "embedding-sdk";
+import { printUsageProblemToConsole } from "embedding-sdk/lib/print-usage-problem";
+import { getSdkUsageProblem } from "embedding-sdk/lib/usage-problem";
+import { setUsageProblem } from "embedding-sdk/store/reducer";
+import { useSetting } from "metabase/common/hooks";
+import { useSelector } from "metabase/lib/redux";
+import { getApplicationName } from "metabase/selectors/whitelabel";
+import { getTokenFeature } from "metabase/setup/selectors";
+
+export function useSdkUsageProblem(config: SDKConfig) {
+  const { allowConsoleLog = true } = config;
+
+  const hasLoggedRef = useRef(false);
+  const appName = useSelector(getApplicationName);
+
+  const dispatch = useDispatch();
+
+  // When the setting haven't been loaded or failed to query, we assume that the
+  // feature is _enabled_ first. Otherwise, when a user's instance is temporarily down,
+  // their customer would see an alarming error message on production.
+  // TODO: replace this with "enable-embedding-sdk" once the feature flag PR landed.
+  const isEnabled = useSetting("enable-embedding") ?? true;
+
+  const hasTokenFeature = useSelector(state => {
+    // We also assume that the feature is enabled if the token-features are missing.
+    // Same reason as above.
+    if (!state.settings.values?.["token-features"]) {
+      return true;
+    }
+
+    // TODO: replace this with "embedding-sdk" once the token feature PR landed.
```

Comment:
as I've mentioned before, I think you should cherry-pick the token feature change and not wait for the feature flag PR, since we already update all token to have `embeddin-sdk`.

That feature flag feature is more about other settings and admin settings change.

Reply:
@WiNloSt I noticed that the `embedding-sdk` flag is the only one with dash-case, while other flags are all underscore_case. Should we fix this?

```tsx
export const tokenFeatures = [
  "attached_dwh",
  "advanced_permissions",
  "audit_app",
  "cache_granular_controls",
  "disable_password_login",
  "content_verification",
  "embedding",
  "embedding-sdk",
  "hosting",
] as const;
```

Reply:
Answered in our DM that the quoted code is outdated, it should be lowercase `embedding_sdk`